packages/messaging/native.js

```diff
@@ -0,0 +1,210 @@
+/**
+ * @module Messaging Implementation Guide
+ * @category Libraries
+ * @description
+ *
+ * Messages will be **sent** to native platforms in 2 formats - notifications and requests.
+ * Notifications do not require a response, but requests do. The following spec explains the difference and
+ * how to handle each.
+ *
+ * ```javascript
+ * // notifications
+ * messaging.notify("helloWorld", { some: "data" })
+ *
+ * // requests
+ * await messaging.request("helloWorld", { some: "data" })
+ * ```
+ *
+ * ## Step 1) Receiving a notification or request message:
+ *
+ * Each platform will 'receive' messages according to their own best practices, the following spec describes everything **after**
+ * the message has been delivered from the clientside JavaScript (deliberately avoiding the platform specifics of *how* messages arrive)
+ *
+ * For example, in Android this would be what happens within a `@Javascript` Interface, but on macOS it would be within
+ * the WebKit messaging protocol, etc.
+ *
+ * ## Algorithm
+ *
+ * 1. let `s` be an incoming raw `JSON` payload
+ * 2. let `msg` be the result of parsing `s` into key/value pairs
+ *     - 2.1 Note: 'parsing' here may not be required if the platform in question receives JSON data directly (ie: JavaScript environments)
+ * 3. if parsing was not successful, throw an "invalid message" Exception
```

Comment:
Are this and the subsequent exceptions delivered back to the sender?

Reply:
I'll think about how to best explain this, in a follow-up :)